platform/viewer/src/lib/DownloadViewportEngine.js

```diff
@@ -0,0 +1,137 @@
+const DOWNLOAD_ELEMENT_ID = 'download-element';
+const PREVIEW_ELEMENT_ID = 'preview-element';
+
+class DownloadViewportEngine {
+  constructor() {
+    this.$previewElement = null;
+    this.$downloadElement = document.createElement('div');
+    this.$activeViewport = null;
+    this.reRenderActionCache = null;
+    this.showAnnotations = false;
+
+    // TODO - Maybe there is a way to get all these tools dynamically?
+    this.availableTools = [
+      'LengthTool',
+      'WwwcTool',
+      'BidirectionalTool',
+      'AngleTool',
+      'StackScrollTool',
+      'BrushTool',
+      'FreehandMouseTool',
+      'EllipticalRoiTool',
+      'CircleRoiTool',
+      'RectangleRoiTool'
+    ];
+
+    this.updateHash = null;
+
+    this.mountPreview = this.mountPreview.bind(this);
+    this.save = this.save.bind(this);
+    this.setElementSize = this.setElementSize.bind(this);
+    this.toggleAnnotations = this.toggleAnnotations.bind(this);
+    this.updateCache = this.updateCache.bind(this);
+    this.enableCornerstoneTools = this.enableCornerstoneTools.bind(this);
+    this.showPreview = this.showPreview.bind(this);
+  }
+
+  setElementSize(element, prop, value) {
+    const resizingElement = element ? element : this.$downloadElement;
+    resizingElement.style[prop] = `${value}px`;
+    const canvas = resizingElement.querySelector('canvas');
+    canvas.style[prop] = `${value}px`;
+  }
+
+  toggleAnnotations() {
+    const element = this.$downloadElement;
+    this.availableTools.forEach(tool => {
+      if (this.showAnnotations) {
+        cornerstoneTools.setToolEnabledForElement(element, tool.replace('Tool', ''));
+      } else {
+        cornerstoneTools.setToolDisabledForElement(element, tool.replace('Tool', ''));
+      }
+    });
+  }
```

Comment:
@swederik, why would someone want an image without annotations? They already have that? If they really want an image without annotations, they could use this tool before adding annotations?

@romulobordezani, that's a dangerous forEach loop to make. We could probably cache that image's `imageIdSpecificToolState`, delete it, take the image, restore it.

Reply:
They might want to zoom in to a specific part of an image. It's definitely less common but still a requirement.

Reply:
Proccess to cache that image's imageIdSpecificToolState, delete it, take the image, restore it -> Done.